File: Go/2215. Find the Difference of Two Arrays.go
package main

//goland:noinspection GoUnusedFunction
func findDifference(nums1 []int, nums2 []int) [][]int {
	set1 := make(map[int]struct{})
	set2 := make(map[int]struct{})

	for _, num := range nums1 {
		set1[num] = struct{}{}
	}

	for _, num := range nums2 {
		set2[num] = struct{}{}
	}

	result := [][]int{{}, {}}

	for num := range set1 {
		if _, exists := set2[num]; !exists {
			result[0] = append(result[0], num)
		}
	}

	for num := range set2 {
		if _, exists := set1[num]; !exists {
			result[1] = append(result[1], num)
		}
	}

	return result
}

File: TypeScript/2215. Find the Difference of Two Arrays.ts
// noinspection JSUnusedGlobalSymbols
const findDifference = (nums1: number[], nums2: number[]): number[][] => {
  const num1 = new Set(nums1)
  const num2 = new Set(nums2)

  const result: number[][] = [[], []]

  for (const num of num1) {
    if (!num2.has(num)) {
      result[0].push(num)
    }
  }

  for (const num of num2) {
    if (!num1.has(num)) {
      result[1].push(num)
    }
  }

  return result
}
